File: javascript/src/problems1_100/remove-duplicates-from-sorted-list.js
class Problem {
  deleteDuplicates(head) {
    throw new Error(`Not implemented (${head}).`);
  }
}

class Default extends Problem {
  deleteDuplicates(head) {
    let current = head;
    while (current && current.next) {
      if (current.val === current.next.val) {
        current.next = current.next.next;
      } else {
        current = current.next;
      }
    }
    return head;
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;

File: python/src/problems1_100/remove_duplicates_from_sorted_list.py
from abc import ABC, abstractmethod
from typing import override

from src.concepts.singly_list_node import SinglyListNode


class Problem(ABC):
    @abstractmethod
    def delete_duplicates(self, head: SinglyListNode | None) -> SinglyListNode | None:
        pass


class Default(Problem):
    @override
    def delete_duplicates(self, head: SinglyListNode | None) -> SinglyListNode | None:
        current: SinglyListNode | None = head
        while current and current.next:
            if current.val == current.next.val:
                current.next = current.next.next
            else:
                current = current.next
        return head


SOLUTIONS: list[Problem] = [Default()]
